### DISCUSSION TITLE
Make loss function configurable

### DISCUSSION THREAD

**Discussion by vv234** [ThilinaRajapakse/simpletransformers#1057] [Ideas]
Hi, 
It seems that I cannot choose my own type of loss functions for the classification models. 
Quickly scanning the src codes for all the models, it looks to me it should be easy to make the loss function configurable.
Moreover, the last sections of all the model classes which apply the loss functions can be pulled out to the base class which would make the configurable loss functions much easier.

Just a suggestion. I may misunderstood the part of the src codes about the loss functions.

Thanks!

**Comment by Zhylkaaa** [comment]
I guess it's not that easy. Moreover as far as I know loss is returned by the model itself (which is a part of transformers library). 
Also, I guess this feature will be hard to maintain, because of models that use custom loss function already (deberta) and different tasks.
So I guess you can just fork transformers and modify it to test custom functions.

**Comment by Zhylkaaa** [comment]
Actually I made a fool of myself :) 
Check #1209

**Comment by vv234** [comment]
That's great, thanks for the update!I will give it a try later.
Jay
    On Sunday, July 25, 2021, 03:33:01 PM PDT, Zhylko Dima ***@***.***> wrote:  
 
 


Actually I made a fool of myself :)
Check #1209
